lib/collection/src/shards/replica_set/mod.rs

```diff
@@ -777,6 +780,61 @@ impl ShardReplicaSet {
         Ok(())
     }
 
+    pub async fn cleanup_local_shard(&self, filter: Filter) -> CollectionResult<UpdateResult> {
+        let local_shard_guard = self.local.read().await;
+
+        let Some(local_shard) = local_shard_guard.deref() else {
+            return Err(CollectionError::bad_request("TODO"));
+        };
+
+        let mut next_offset = Some(ExtendedPointId::NumId(0));
+        let mut ids = Vec::new();
+
+        while let Some(current_offset) = next_offset {
+            let mut points = local_shard
+                .get()
+                .scroll_by(
```

Comment:
I don't see parameter names in the GitHub UI, but it looks like we're using different limits: 100 and 1k

Reply:
Nice catch, thx.